Packs/MicrosoftECM/Integrations/MicrosoftECM/MicrosoftECM.ps1

```diff
@@ -0,0 +1,1074 @@
+
+.OUTPUTS
+Return the used parameters or throws an exception if parameter set cannot be resolved
+#>
+Function ValidateCreateScriptParams($script_file_entry_id, $script_text)
+{
+	AssertNoMoreThenOneParameterGiven "script_file_entry_id cannot be resolved with script_text" $script_file_entry_id $script_text
+	if (!$script_file_entry_id -And !$script_text)
+	{
+		throw "Please supply either script_file_entry_id or script_text"
+	}
+	if ($script_file_entry_id)
+	{
+		return "script_path"
+	}
+	return "script_text"
+}
+<#
+.DESCRIPTION
+This function Verifies a valid parameter set is used with excactly one of $collection_id $collection_name and one of $include_collection_id $include_collection_name
+For more info see https://docs.microsoft.com/en-us/powershell/module/configurationmanager/add-cmdevicecollectionincludemembershiprule?view=sccm-ps
+
+.PARAMETER collection_id
+Specifies the collection ID
+
+.PARAMETER collection_name
+Specifies the collection name
+
+.PARAMETER collection_id
+Specifies the collection ID to include\exclude in the membership rule
+
+.PARAMETER collection_name
+Specifies the collection name to include\exclude in the membership rule
+
+.OUTPUTS
+Return the used parameters or throws an exception if parameter set cannot be resolved
+#>
+Function ValidateIncludeOrExcludeDeviceCollectionParameters($collection_id, $collection_name, $include_collection_id, $include_collection_name)
+{
+	AssertNoMoreThenOneParameterGiven "Can only use one of the following parameters: collection_name, collection_id" $collection_name $collection_id
+	if (!$collection_id -And !$collection_name)
+	{
+		throw "Must use one of the following parameters: collection_id, collection_name"
+	}
+	AssertNoMoreThenOneParameterGiven "Can only use one of the following parameters: include\exclude_collection_name, include\exclude_collection_id" $include_collection_name $include_collection_id
+	if (!$include_collection_id -And !$include_collection_name)
+	{
+		throw "Must use one of the following parameters: include\exclude_collection_id, include\exclude_collection_name"
+	}
+	if ($collection_id -And $include_collection_id)
+	{
+		return "id&id"
+	}
+	if ($collection_id -And $include_collection_name)
+	{
+		return "id&name"
+	}
+	if ($collection_name -And $include_collection_id)
+	{
+		return "name&id"
+	}
+	if ($collection_name -And $include_collection_name)
+	{
+		return "name&name"
+	}
+}
+<#
+.DESCRIPTION
+This function Parses A configuration manager collection object into PSCustomObject with selected keys.
+
+.PARAMETER collection
+Specifies collection to parse
+
+.OUTPUTS
+Return the PSCustomObject with the selected collection keys
+#>
+Function ParseCollectionObject($collection)
+{
+	[PSCustomObject]@{
+		Name = $collection.Name
+		ID = $collection.CollectionID
+		Type = $COLLECTION_TYPE_MAPPING["$( $collection.CollectionType )"]
+		Comment = $collection.Comment
+		CurrentStatus = $COLLECTION_CURRENT_STATUS_MAPPING["$( $collection.CurrentStatus )"]
```

Comment:
`someproperty = $hashtable.Get_Item($collection.CollectionType)`
You could try this instead, should work.